core/src/main/java/com/nisovin/magicspells/mana/ManaSystem.java

```diff
@@ -13,44 +18,45 @@
 
 public class ManaSystem extends ManaHandler {
 
-	private String defaultBarPrefix;
-	private char defaultSymbol;
-	private int defaultBarSize;
-	private ChatColor defaultBarColorFull;
-	private ChatColor defaultBarColorEmpty;
-
-	private int defaultMaxMana;
-	private int defaultStartingMana;
-	private int defaultRegenAmount;
-	private int defaultRegenInterval;
-
-	private boolean showManaOnUse;
-	private boolean showManaOnRegen;
-	private boolean showManaOnHungerBar;
-	private boolean showManaOnActionBar;
-	private boolean showManaOnExperienceBar;
+	private final boolean showManaOnUse;
+	private final boolean showManaOnRegen;
+	private final boolean showManaOnHungerBar;
+	private final boolean showManaOnActionBar;
+	private final boolean showManaOnExperienceBar;
 
 	private List<String> modifierList;
 	private ModifierSet modifiers;
-	
-	private ManaRank defaultRank;
-	private List<ManaRank> ranks;
-	
-	private Map<UUID, ManaBar> manaBars;
-	private Set<Regenerator> regenerators;
+
+	private final ManaRank defaultRank;
+	private final List<ManaRank> ranks;
+
+	private final Map<UUID, ManaBar> manaBars;
+	private final Set<Regenerator> regenerators;
 
 	public ManaSystem(MagicConfig config) {
 		String path = "mana.";
-		defaultBarPrefix = config.getString(path + "default-prefix", "Mana:");
-		defaultSymbol = config.getString(path + "default-symbol", "=").charAt(0);
-		defaultBarSize = config.getInt(path + "default-size", 35);
-		defaultBarColorFull = ChatColor.getByChar(config.getString(path + "default-color-full", ChatColor.GREEN.getChar() + ""));
-		defaultBarColorEmpty = ChatColor.getByChar(config.getString(path + "default-color-empty", ChatColor.BLACK.getChar() + ""));
 
-		defaultMaxMana = config.getInt(path + "default-max-mana", 100);
-		defaultStartingMana = config.getInt(path + "default-starting-mana", defaultMaxMana);
-		defaultRegenAmount = config.getInt(path + "default-regen-amount", 5);
-		defaultRegenInterval = config.getInt(path + "default-regen-interval", TimeUtil.TICKS_PER_SECOND);
+		String legacyDefaultBarPrefix = Util.getStrictString(config.getString(path + "default-prefix", "Mana:"));
+		NamedTextColor legacyDefaultColorFull = Util.getLegacyColor(config.getString(path + "default-color-full", null), NamedTextColor.GREEN);
+		NamedTextColor legacyDefaultColorEmpty = Util.getLegacyColor(config.getString(path + "default-color-empty", null), NamedTextColor.BLACK);
+
+		String defaultBarFormat = config.getString(path + "default-bar-format", null);
```

Comment:
`mana.yml` seems to be out of date with some of these changes. It still has a new `style-full` & `style-empty` options which don't appear to be read from here, while it doesn't have `bar-format`.